### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by jparise**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      if: matrix.python-version == '2.7'
+      run: |
+        python -m pip install --upgrade pip
+        pip install mock
+    - name: Tests
+      run: |
+        python setup.py test
```
Comment (elprans) [inline .github/workflows/ci.yml:31]: `sudo` might still be required here

Comment (jparise) [reply .github/workflows/ci.yml:31]: That didn't seem to help. 😞

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      if: matrix.python-version == '2.7'
+      run: |
+        python -m pip install --upgrade pip
+        pip install mock
```
Comment (elprans) [inline .github/workflows/ci.yml:28]: Remove "mock" from `tests_require` in setup.py, I think that's what trips some of the tests up.

**Comment by jparise** [comment]
I understand the test failures now. They're failing because the current Python + OS environment results in `openat` syscalls when Python's `open()` is used.

I'll work on updating the tests.

**Review by elprans — APPROVED** [review]
> They're failing because the current Python + OS environment results in `openat` syscalls when Python's `open()` is used.

Yep, good find! Glibc switched to using the `openat` syscall for `open()` by default [in 2.26](https://sourceware.org/git/?p=glibc.git;a=commit;h=b41152d716ee9c5ba34495a54e64ea2b732139b5)

**Pull request merged**